core/src/test/scala/pickling/run/self-describing-test.scala

```diff
@@ -0,0 +1,63 @@
+package scala.pickling.runtime
+
+import org.scalatest.FunSuite
+import scala.pickling.{Ops, Pickler, Unpickler}
+import scala.pickling.pickler.AllPicklers
+import scala.pickling.binary.{BinaryFormats, BinaryPickleArray}
+import scala.pickling.internal.{HybridRuntime}
+
+abstract class SDPicklingLogic extends Ops with AllPicklers with BinaryFormats
+
+object SDPicklingProtocol extends {
+  val ignoreMe = scala.pickling.internal.replaceRuntime(new HybridRuntime)
```

Comment:
It is really necessary that `HybridRuntime` is enabled, otherwise it will use runtime reflection to unpickle.